### PR TITLE
feat: new package header

### LINKED ISSUE #349: Redesign of package header
Proposed redesign of the package header to highlight some new information, a more clear notification when you are viewing a non-latest version of a package, and some information hierarchy improvements.

[Figma link
](https://www.figma.com/file/UpBSyDWbD2QUqYzCJ6eegq/Doc-Formatting?type=design&node-id=88%3A4000&mode=dev&t=5sSWca7Jcs72rZ9v-1)

image

### PR DESCRIPTION
**Pull request by crowlKats**
Closes #349

### CODE REVIEW

**Review by marvinhagemeister — APPROVED** [review]

**Comment by lucacasonato** [comment]
At some viewport sizes, this bug occurs:

**Comment by lucacasonato** [comment]
On mobile there is some incorrect spacing under the description here: 
image

File: frontend/routes/package/(_components)/PackageHeader.tsx [hunk 1]
```diff
@@ -13,81 +17,135 @@ export function PackageHeader(
...
-              <div
-                class={`flex w-12 h-12 items-center justify-center aspect-square rounded-full p-1 ${
-                  getScoreBgColorClass(pkg.score)
-                }`}
-                style={`background-image: conic-gradient(transparent, transparent ${pkg.score}%, #e7e8e8 ${pkg.score}%)`}
+
+        <div class="flex md:items-end flex-col gap-2 md:gap-4 text-right pb-4">
+          <div class="flex flex-col md:flex-row gap-2 md:gap-8 items-between">
+            <div class="flex flex-row md:flex-col items-center md:items-end gap-2 md:gap-1 text-sm font-bold">
+              <div>Works With</div>
```
Comment (lucacasonato) [inline frontend/routes/package/(_components)/PackageHeader.tsx:113]:
```suggestion
              <div>Works with</div>
```

**Comment by lucacasonato** [comment]
There is a spacing issue to the left of the "Jump to latest" button:

**Comment by lucacasonato** [comment]
Looking great - just some small comments.

**Review by lucacasonato — APPROVED** [review]
Looks good!

**Pull request merged**